src/g/selection/SelectionCol.js

```diff
@@ -200,34 +225,328 @@ const SelectionManager = Collection.extend({
+
+    const selectionType = selection.get("type");
+    const selSeqIdIdx = seqIdToIdxMap[selection.get("seqId")];
+    const selXStart = selection.get("xStart");
+    const selXEnd = selection.get("xEnd");
+
+    const minXStart = Math.min(lSelXStart, selXStart);
+    const maxXEnd = Math.max(lSelXEnd, selXEnd);
+    const minSeqIdIdx = Math.min(lSelSeqIdIdx, selSeqIdIdx);
+    const maxSeqIdIdx = Math.max(lSelSeqIdIdx, selSeqIdIdx);
+
+    if (lastSelectionType === "row" && selectionType === "row") {
+      // Select all rows between the last selection and the current selection
+      const selections = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        selections.push(new rowsel({seqId: idxToSeqIdMap[i]}));
+      }
+      this.add(this._getSelsWithLabelsForRows(selections), {silent: true});
+    } else if (lastSelectionType === "column" && selectionType === "column") {
+      // Select all columns between the last selection and the current selection
+      const columns = [];
+      for (let i = minXStart; i <= maxXEnd; i++) {
+          columns.push(new columnsel({xStart: i, xEnd: i}));
+      }
+      this.add(columns, {silent: true});
+    } else if (lastSelectionType === "pos" && selectionType === "pos" ) {
+      // Select all residues between the last selection and the current selection
+      const positions = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        for (let j = minXStart; j <= maxXEnd; j++) {
+          positions.push(new possel({xStart: j, xEnd: j, seqId: idxToSeqIdMap[i]}));
+        }
+      }
+      this.add(positions, {silent: true});
+    } else if (lastSelectionType === "label" && selectionType === "label" ) {
+      // Select all labels between the last selection and the current selection
+      const labels = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        labels.push(new labelsel({seqId: idxToSeqIdMap[i]}));
+      }
+      this.add(labels, {silent: true});
+    } else if (lastSelectionType === "row" && selectionType === "pos") {
+      const positions = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        for (let j = selXStart; j <= selXEnd; j++) {
+          positions.push(new possel({xStart: j, xEnd: j, seqId: idxToSeqIdMap[i]}));
         }
       }
+      this.add(positions, {silent: true});
+    } else if (lastSelectionType === "column" && selectionType === "pos") {
+      const positions = [];
+      for (let j = minXStart; j <= maxXEnd; j++) {
+        positions.push(new possel({xStart: j, xEnd: j, seqId: idxToSeqIdMap[selSeqIdIdx]}));
+      }
+      this.add(positions, {silent: true});
     } else {
-      if (e.ctrlKey || e.metaKey) {
-        this.add(selection);
+      // Select the current selection
+      this.add(this._getSelsWithLabelsForRows([selection]), {silent: true});
+    }
+  },
+
+  _isAlreadySelected: function(selection) {
+    // Models in the collection are refined selections
```

Comment:
We won't need to maintain this assumption in the future if we dispense with the **range** column and position selections. We don't even require range column or position selections.

Reply:
Currently, we are running `_refineSelections()` after each selection to break down contiguous selections. So, it is not an assumption, but definitely an overhead.

Reply:
Can you file a tech debt ticket to break the refactor the existing selection models so that they don't handle ranges?
We likely won't work on it, but it would be useful to note it down so that once we rewrite MSA we remember this.

Reply:
Done, https://schrodinger.atlassian.net/browse/LDTOOLS-166.